internal/services/loganalytics/log_analytics_workspace_table_custom_log_resource.go

```diff
@@ -0,0 +1,368 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package loganalytics
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/operationalinsights/2022-10-01/tables"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/operationalinsights/2022-10-01/workspaces"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var (
+	_ sdk.ResourceWithUpdate        = WorkspaceTableCustomLogResource{}
+	_ sdk.ResourceWithCustomizeDiff = WorkspaceTableCustomLogResource{}
+)
+
+type WorkspaceTableCustomLogResource struct{}
+
+type WorkspaceTableCustomLogResourceModel struct {
+	Name                 string                 `tfschema:"name"`
+	WorkspaceId          string                 `tfschema:"workspace_id"`
+	DisplayName          string                 `tfschema:"display_name"`
+	Description          string                 `tfschema:"description"`
+	Plan                 string                 `tfschema:"plan"`
+	Columns              []workspaceTableColumn `tfschema:"column"`
+	Solutions            []string               `tfschema:"solutions"`
+	StandardColumns      []workspaceTableColumn `tfschema:"standard_column"`
+	RetentionInDays      int64                  `tfschema:"retention_in_days"`
+	TotalRetentionInDays int64                  `tfschema:"total_retention_in_days"`
+}
+
+func (r WorkspaceTableCustomLogResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var table WorkspaceTableCustomLogResourceModel
+			if err := metadata.DecodeDiff(&table); err != nil {
+				return err
+			}
+
+			if table.Plan == string(tables.TablePlanEnumBasic) {
+				if _, ok := metadata.ResourceDiff.GetOk("retention_in_days"); ok {
```

Comment:
Should we also check `total_retention_in_days` here as well since we're not sending it if it's a Basic plan down below.